### ISSUE TITLE
do you need to run it from the ssllabs-scan directory?

### ISSUE THREAD

**Issue by gmmann** [kyhau/ssllabs-scan#4]
This is great and so far useful. 
I'm running it from the subdirectory(of the ssllabs-scan directory) that the input file is in and it is odd that working for some in the input file list and getting these errors for others in the list
i didn't want to run it in the main directory as trying to scan over 3000 sites

File "$BASEDIR/ssllabs-scan/bin/ssllabs-scan", line 11, in <module>
    load_entry_point('ssllabsscan', 'console_scripts', 'ssllabs-scan')()
  File "$BASEDIR/ssllabs-scan/ssllabsscan/main.py", line 82, in main
    return process(server_list_file=sys.argv[1])
  File "$BASEDIR/ssllabs-scan/ssllabsscan/main.py", line 68, in process
    traceback.print_stack()

**Comment by kyhau** [comment]
> do you need to run it from the ssllabs-scan directory?

No, you can run the `ssllabs-scan` binary from any location/folder, if you have activated the virtualenv as detailed in the README.md. Alternatively, if you want to run it without activating the virtualenv after you built it, then use full path `/path-to/ssllabs-scan/env/bin/ssllabs-scan`.

After you run `pip install` to build the code and binary,  the binary `ssllabs-scan` is created in the virtualenv you specified, e.g. `env/bin/ssllabs-scan`, and its path is set within your virtualenv, such that you can simply call `ssllabs-scan` from anywhere you like.

You can double check where is `ssllabs-scan` by running after activating your virtualenv.

```
which ssllabs-scan
```

> I'm running it from the subdirectory(of the ssllabs-scan directory) that the input file is in

`ssllabs-scan` takes a simple file (list of server urls) as input - the file does not need to be in the same location as the binary `ssllabs-scan` or the source file `main.py`. You can specify the location to the file (as mentioned in the README.md). E.g.

`ssllabs-scan path-to/your_server_list.txt`

> it is odd that working for some in the input file list and getting these errors for others in the list

I have not seen this error before. You mentioned about "over 3000 sites". I'm not sure if it is related to 
the "Access Rate and Rate Limiting" detailed in https://github.com/ssllabs/ssllabs-scan/blob/stable/ssllabs-api-docs.md. 

Hope this help.

**Comment by gmmann** [comment]
thanks.  i do have it running in a virtual environment and it was odd that even on the 3rd scan entry in the list file,  it came up with those messages.  I'm just trying to make sure if it was anything i missed as seems random.

**Comment by pastrom** [comment]
Hi,

I'm also thankfull for this work as the other python scrips for this API I've found has been old and out dated.

I looked in to this issue and found that it is occuring due to the code not handling all response status codes. In normal state the server accepts a request and returns response.status_code 200. When the API server is overloaded with requests it returns response.status_code 529.  

One suggestion:
<img width="1326" alt="Screenshot 2020-01-08 at 10 22 16" src="https://user-images.githubusercontent.com/24839398/71965906-dc8f7780-3200-11ea-8b74-07184ccec78c.png">
